core/src/test/java/hudson/FilePathTest.java

```diff
@@ -349,8 +353,44 @@ private FilePath createFilePath(final File base, final String... path) throws IO
         final String filePrefix = "JENKINS-10629";
         checkTarUntarRoundTrip(filePrefix, largeFileSize);
     }
-     
+
+    private void archive2ReturnsPaths(String filePrefix, long fileSize) throws IOException, InterruptedException {
+
+        final File tmpDir = temp.newFolder(filePrefix);
+        tmpDir.mkdirs();
+        final File tempFile =  new File(tmpDir, filePrefix + ".log");
+        RandomAccessFile file = new RandomAccessFile(tempFile, "rw");
+
+        file.setLength(fileSize);
+        assumeTrue(fileSize == file.length());
+        file.close();
+
+        final File tmpDir2 = new File(tmpDir, filePrefix);
+        tmpDir2.mkdirs();
+
+
+        final File tempFile2 =  new File(tmpDir2, filePrefix + ".log");
+        RandomAccessFile file2 = new RandomAccessFile(tempFile2, "rw");
+        file2.setLength(fileSize);
+        assumeTrue(fileSize == file2.length());
+        file2.close();
+
+
+
+        final FilePath tmpDirPath = new FilePath(tmpDir);
+        try (OutputStream os = new FileOutputStream(tempFile)) {
+            DirScanner.Glob scanner = new DirScanner.Glob("**", "");
+            List<String> files = tmpDirPath.archive2(ArchiverFactory.TARGZ, os, scanner);
+            assertThat(files.size(), is(2));
+            assertThat(files.get(1), StringEndsWith.endsWith(format("%s/%s.log", filePrefix,filePrefix)));
```

Comment:
does this work on windows?  (are the names of the files always normalized to `/`  in which case the Javadoc must state this, otherwise it looks like this test will fail